sktime/datatypes/_utilities.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+"""Get index of time series data, helper function."""
+
+import numpy as np
+import pandas as pd
+
+
+def _get_index(x):
+    if hasattr(x, "index"):
+        return x.index
+    else:
+        # select last dimension for time index
+        return pd.RangeIndex(x.shape[-1])
+
+
+def get_time_index(X):
+    """Get index of time series data, helper function.
+
+    Parameters
+    ----------
+    X : pd.DataFrame / pd.Series / np.ndarray
```

Comment:
I would strongly recommend, please write which format assumptions you make here. E.g., which mtypes can this be?
Otherwise it's hard to review, to mentally check cases and assumptions against what is being computed.